app/backend/db/crud.py

```diff
@@ -1,45 +1,54 @@
-PASSWORD = os.environ.get('ONE_REPORT_PASSWORD', 'one_report')
-HOST = os.environ.get('ONE_REPORT_HOST', 'localhost')
-DB = os.environ.get('ONE_REPORT_DB', 'one_report')
-PORT = os.environ.get('ONE_REPORT_PORT', '5432')
-
-DATABASE_URI = f'postgres+psycopg2://{USERNAME}:{PASSWORD}@{HOST}:{PORT}/{DB}'
-
-engine = create_engine(DATABASE_URI)
-Session = sessionmaker(bind=engine)
-
-
-@contextmanager
-def transaction():
-    s = Session()
-    try:
-        yield s
-
-    except:
-        s.rollback()
-        raise
-
-    finally:
-        s.close()
-
-
-def recreate_database():
-    with transaction() as s:
-        models.Base.metadata.drop_all(engine)
-        models.Base.metadata.create_all(engine)
-        mador = models.Mador(name='pie')
-        mador_settings = models.MadorSettings(
-            mador=mador, key='default_reminder_time', value='08:00', type='time')
-
-        s.add_all([mador, mador_settings])
-        s.commit()
+from sqlalchemy.orm import Session
+from datetime import date, time
+from typing import List
+
+from .models import User, DateData
+from utils.datetime_utils import daterange
+from db import schemas
+
+# User:
+
+def get_user(db: Session, user_id: int) -> User:
+    return db.query(User).filter(User.id == user_id).one()
+
+def get_subjects(db: Session, commander_id: int) -> List[User]:
+    commander = get_user(db, commander_id)
+    return commander.soldiers
+
+def get_reminder(db: Session, user_id:int) -> time:
+    user = get_user(db, user_id)
+    return user.reminder_time
+
+def was_reminded(db: Session, user_id: int) -> bool:
+    user = get_user(db, user_id)
+    return user.last_reminded_date == date.today()
+
+# Date Data:
+
+def get_dates_data(db: Session, user_id: int, 
+                   start_date: date, end_date: date = None) -> List[DateData]:
+    return db.query(DateData).filter(DateData.user_id == user_id, 
+                                     start_date <= DateData.date <= end_date).all()
+
+def get_reduced_dates_data(db: Session, user_id: int, 
+                           start_date: date, end_date: date = None) -> List[schemas.DateData]:
+    """Return list of DateData after adding follow same dates into range."""
+    dates_data = get_dates_data(db, user_id, start_date, end_date)
+    responce = []
+    for i in range(len(dates_data)):
+        if i > 0 and dates_data[-1] == dates_data:
+            responce[-1].end_date = dates_data[i].date
+        
+        else:
+            responce.append(schemas.DateData(user=dates_data[i].user,
+                                             start_date=dates_data[i].date, 
+                                             state=dates_data[i].state,
+                                             reported_by=dates_data[i].reported_by,
+                                             reason=dates_data[i].reason,
+                                             date_details=dates_data[i].date_details))
+    return responce
+
```

Comment:
this conversion to response Scheme shouldnt be in crud.
it should be in the backend/serialize or somewhere like that

Reply:
I did it more for the object who has date range and is already implemented, over writing another kinda useless class

Reply:
still..
the db shouldn't import anything from schemas and schemas should go to backend.
DB doesn't know anything about backend.
only the backend knows about backend :)

Reply:
i dont agree about "the db shouldn't import anything from schemas"
you can see there:
https://fastapi.tiangolo.com/tutorial/sql-databases/
it imports the scheme for  function's input/output types:
![image](https://user-images.githubusercontent.com/31568827/71423831-b1e7df80-2694-11ea-8784-414074c95cc9.png)

Reply:
for using the model i can change, tho it will be kida of  a code duplicate